src/laser/Laser.cpp

```diff
@@ -35,70 +55,671 @@ Laser::InitData (const amrex::BoxArray& slice_ba,
+                    lapR = i>imin && i<imax && j>jmin && j<jmax ?
+                        (n00j00_arr(i+1,j,0)+n00j00_arr(i-1,j,0)-2._rt*n00j00_arr(i,j,0))/(dx*dx) +
+                        (n00j00_arr(i,j+1,0)+n00j00_arr(i,j-1,0)-2._rt*n00j00_arr(i,j,0))/(dy*dy) : 0._rt;
+                    lapI = i>imin && i<imax && j>jmin && j<jmax ?
+                        (n00j00_arr(i+1,j,1)+n00j00_arr(i-1,j,1)-2._rt*n00j00_arr(i,j,1))/(dx*dx) +
+                        (n00j00_arr(i,j+1,1)+n00j00_arr(i,j-1,1)-2._rt*n00j00_arr(i,j,1))/(dy*dy) : 0._rt;
+                } else {
+                    lapR = i>imin && i<imax && j>jmin && j<jmax ?
+                        (nm1j00_arr(i+1,j,0)+nm1j00_arr(i-1,j,0)-2._rt*nm1j00_arr(i,j,0))/(dx*dx) +
+                        (nm1j00_arr(i,j+1,0)+nm1j00_arr(i,j-1,0)-2._rt*nm1j00_arr(i,j,0))/(dy*dy) : 0._rt;
+                    lapI = i>imin && i<imax && j>jmin && j<jmax ?
+                        (nm1j00_arr(i+1,j,1)+nm1j00_arr(i-1,j,1)-2._rt*nm1j00_arr(i,j,1))/(dx*dx) +
+                        (nm1j00_arr(i,j+1,1)+nm1j00_arr(i,j-1,1)-2._rt*nm1j00_arr(i,j,1))/(dy*dy) : 0._rt;
+                }
+                const Complex lapA = lapR + I*lapI;
+                const Complex an00j00 = n00j00_arr(i,j,0) + I * n00j00_arr(i,j,1);
+                const Complex anp1jp1 = np1jp1_arr(i,j,0) + I * np1jp1_arr(i,j,1);
+                const Complex anp1jp2 = np1jp2_arr(i,j,0) + I * np1jp2_arr(i,j,1);
+                Complex rhs;
+                if (step == 0) {
+                    // First time step: non-centered push to go
+                    // from step 0 to step 1 without knowing -1.
+                    const Complex an00jp1 = n00jp1_arr(i,j,0) + I * n00jp1_arr(i,j,1);
+                    const Complex an00jp2 = n00jp2_arr(i,j,0) + I * n00jp2_arr(i,j,1);
+                    rhs =
+                        + 8._rt/(c*dt*dz)*(-anp1jp1+an00jp1)*edt1
+                        + 2._rt/(c*dt*dz)*(+anp1jp2-an00jp2)*edt2
+                        - 2._rt * chi_fac * isl_arr(i,j,chi) * an00j00
+                        - lapA
+                        + ( -6._rt/(c*dt*dz) + 4._rt*I*djn/(c*dt) + I*4._rt*k0/(c*dt) ) * an00j00;
+                } else {
+                    const Complex anm1jp1 = nm1jp1_arr(i,j,0) + I * nm1jp1_arr(i,j,1);
+                    const Complex anm1jp2 = nm1jp2_arr(i,j,0) + I * nm1jp2_arr(i,j,1);
+                    const Complex anm1j00 = nm1j00_arr(i,j,0) + I * nm1j00_arr(i,j,1);
+                    rhs =
+                        + 4._rt/(c*dt*dz)*(-anp1jp1+anm1jp1)*edt1
+                        + 1._rt/(c*dt*dz)*(+anp1jp2-anm1jp2)*edt2
+                        - 4._rt/(c*c*dt*dt)*an00j00
+                        - 2._rt * chi_fac * isl_arr(i,j,chi) * an00j00
+                        - lapA
+                        + ( -3._rt/(c*dt*dz) + 2._rt*I*djn/(c*dt) + 2._rt/(c*c*dt*dt) + I*2._rt*k0/(c*dt) ) * anm1j00;
+                }
+                rhs_arr(i,j,0) = rhs;
+            });
```

Comment:
Up to here it is the same as in the MG solver, should we outsource this in some preparation function?

Reply:
There are actually some differences. We could abstract them away, but a bit more tests would probably be needed to see what is most useful on the long run. Therefore, I would like to keep it that way.